scripts/analysis.py

```diff
@@ -0,0 +1,378 @@
+
+    nuclide_set = set(nuclides)
+    mass_dict = collections.OrderedDict({})
+
+    for nuclide in nuclide_set:
+        temp_nuclide_sum = 0 
+        for i in range(len(nuclides)):
+            if nuclides[i] == nuclide:
+                temp_nuclide_sum += mass_of_nuclides[i]
+        mass_dict[nuclide] = temp_nuclide_sum
+
+
+    for nuclide in mass_dict:
+        nuclide_name = str(nucname.name(nuclide))
+        nuclide_inven += nuclide_name + ' = ' + str(mass_dict[nuclide]) + 'kg \n'
+
+    return nuclide_inven
+
+
+def capacity_calc(governments, timestep, entry, exit):
+    """ Adds and subtracts capacity over time for plotting
+
+    Paramters
+    ---------
+    governments: array
+        array of governments (countries)
+    timestep: array
+        array of timestep from 0 to simulation time
+    entry: array
+        power_cap, agentid, parentid, entertime
+        of all entered reactors
+    exit: array
+        power_cap, agentid, parenitd, exittime
+        of all decommissioned reactors
+
+    Returns
+    -------
+    power_dict: dictionary
+        dictionary of capacity progression with country_government as key
+    num_dict: dictionary
+        dictionary of number of reactors progression
+        with country_government as key
+    """
+
+    power_dict = collections.OrderedDict({})
+    num_dict = collections.OrderedDict({})
+
+    for gov in governments:
+        capacity = []
+        num_reactors = []
+        cap = 0
+        count = 0
+        for num in timestep:
+            for enter in entry:
+                if enter[3] == num and enter[2] == gov[1]:
+                    cap += enter[0]
+                    count += 1
+            for dec in exit:
+                if dec[3] == num and dec[2] == gov[1]:
+                    cap -= dec[0]
+                    count -= 1
+            capacity.append(cap)
+            num_reactors.append(count)
+
+        power_dict[gov[0]] = np.asarray(capacity)
+        num_dict[gov[0]] = np.asarray(num_reactors)
+
+    return power_dict, num_dict
+
+
+def stacked_bar_chart(dictionary, timestep, xlabel, ylabel, title, outputname):
+    """ Creates stacked bar chart of timstep vs dictionary
+
+    Parameters
+    ----------
+    dictionary: dictionary
+        holds time series data
+    timestep: array
+        array of timestep (x axis)
+    xlabel: string
+        xlabel of plot
+    ylabel: string
+        ylabel of plot
+    title: string
+        title of plot
+
+    Returns
+    -------
+    Stacked bar chart
+    """
+
+    # set different colors for each bar
+    color_index = 0
+    top_index = True
+    prev = ''
+    plot_array = []
+    # for every country, create bar chart with different color
+    for key in dictionary:
+        label = key.replace('_government', '')
+        # very first country does not have a 'bottom' argument
+        if top_index is True:
+            plot = plt.bar(1950+(timestep/12), dictionary[key], .5,
```

Comment:
Please note that in python 2.7 this division by an int plus another int will result in an int. 
It does the right thing in python 3, but if someone runs your code with python 2, you'll lose month resolution. 
 
Here is a great example of where you could create a function that does a singular atomic thing, in order to avoid a super duper long function.

```python
def timestep_to_year(timestep=0, startyear=1950, startmonth=0):
    """
    Returns a fractional year (e.g. 1950.5 for June 1950) based on the timestep and start date.

    Parameters
    ----------
    startyear: int
        year at the beginning of the simulation
    timesteps: int
        timestep to convert into year
    startmonth: int
        month of the year, starting with January (0)

    Returns
    ----------
    year : float
        the fractional year, representing the timestep given
    """
    return float(startyear) + (timestep + startmonth)/12.0
```